src/parser.rs

```diff
@@ -119,12 +123,24 @@ assert_eq!(
 
                 if let Some(b'=') = parser.peek() {
                     parser.next();
-                    parser.parse_list_entry(entry_visitor)
+                    if let Some(entry_visitor) = entry_visitor {
```

Comment:
This looks bad, but it is due to the type diversity involved here.  Each fork produces a new type and I didn't want to get into the whole `&dyn` mess for such a small-scale fork.